### ISSUE TITLE
INCLUDE_CHILDREN and FIXED_ORDER

### ISSUE THREAD

**Issue by uruuru** [eclipse/elk#40]
There's an issue with `FIXED_ORDER` port constraints on compound nodes when hierarchy handling is set to `INCLUDE_CHILDREN`.

The following .kgt yields a null pointer due to `portPosition` being `null` which is explicitly passed as `null` via a parameter. 

```
at org.eclipse.elk.alg.layered.graph.LGraphUtil.createExternalPortDummy(LGraphUtil.java:839)
```

```
properties: 
    org.eclipse.elk.hierarchyHandling = INCLUDE_CHILDREN

knode n0 {
    properties:
        org.eclipse.elk.portConstraints = FIXED_ORDER
    knode n1
}
knode n2 {
    kedge (-> n1)
} 
```

**Comment by eNBeWe** [comment]
What is the expected behaviour in this case? (Apart from "no null pointer")
a) If the order of ports is fixed, where should the dummy port be inserted? 
b) Should a dummy port that is introduced during the layout be subjected to the port constraints?

**Comment by le-cds** [comment]
In the case of hierarchical layout, we actually do have a bit of flexibility regarding where dummy nodes for hierarchical ports are placed. The order between hierarchical port dummies is guaranteed by inserting in layer successor constraints. If a hierarchical port dummy was inserted to split a hierarchical edge (which implies that we wouldn't know where to insert it, as eNBeWe commented), we could simply ignore it while generating the constraints. Might work. This would of course allow the port to be inserted at any point between the explicitly defined ports.

**Comment by le-cds** [comment]
Might have been fixed in #160.

**Comment by le-cds** [comment]
> Might have been fixed in #160.

Fixed indeed.
